assets/src/js/Components/ArchivePicker/ArchivePicker.js

```diff
@@ -233,7 +265,50 @@ export default function ArchivePicker() {
     }
   }, [loading, loaded, images, error, searchText, pageNumber, dispatch]);
 
-  const includeInWp = async (ids = []) => {
+  const currentBlock = wp.data.select('core/block-editor').getSelectedBlock();
+
+  const getImageDetails = useCallback(async id => {
+    try {
+      // On first try wp returns undefined, so need to make call again to get image details.
+      for (let retries = 0; retries < 2; ++retries) {
+        const newImageUploaded = await wp.data.select('core').getMedia(id);
+        if (newImageUploaded) {
+          return newImageUploaded;
+        }
+
+        await timeout(3000);
+      }
+    } catch (err) {
+      throw err;
+    }
+  }, []);
+
+  const processImageToAddToEditor = async id => {
+    dispatch({type: ACTIONS.ADDING_IMAGE_TO_POST});
+    try {
+      if (currentBlock.name === 'core/image') {
+        // Get current Image details from WP
+        const uploadedImage = await getImageDetails(id);
+        // Get Updated Atrributes
+        const updatedAttributes = updateImageBlockAttributes(uploadedImage, currentBlock);
+        await wp.data.dispatch('core/block-editor').updateBlock(currentBlock.clientId, updatedAttributes);
+      }
+
+      if (currentBlock.name === 'planet4-blocks/happypoint') {
+        const updatedAttributes = updateHappyPointAttributes(id);
+        await wp.data.dispatch('core/block-editor').updateBlock(currentBlock.clientId, updatedAttributes);
+      }
+
+      dispatch({type: ACTIONS.ADDED_IMAGE_TO_POST});
+      await timeout(1000);
+      dispatch({type: ACTIONS.CLOSE_SIDEBAR});
+      document.querySelector('.media-modal-close').click();
+    } catch (err) {
+      throw err;
```

Comment:
Shouldn't we dispatch an error if it fails? Consider if it's fine to re-use `SET_ERROR` OR `PROCESS_IMAGES_ERROR` OR create a new one like `ADD_IMAGE_TO_POST_ERROR`